File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    dic={};
    let result=new Array(id_list.length).fill(0)
    id_list.forEach(item=>dic[item]=0);
    report=[...new Set(report)]
    report.forEach(item=>{
        item=item.split(" ");
        dic[item[1]]+=1;
    })

    report.forEach(item=>{
        item=item.split(" ");
        if(dic[item[1]]>=k){
            result[id_list.indexOf(item[0])]+=1;
        }
    })
    return result;
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    dic=dict()
    result=[]
    users=[]
    report=list(set(report))
    for item in id_list:
        dic[item]=[]
    for item in report:
        item=item.split(" ")
        dic[item[1]].append(item[0])
    for key, value in dic.items():
        if len(value)>=k:
            users.extend(value)
    for item in id_list:
        result.append(users.count(item))
        
    return result
    